### PR TITLE
Make audit models str method python 2 and 3 compatible

### PR DESCRIPTION
**Pull request by caktus-philip**
This is a re-do of #990, but based on `develop` instead of `prp-refactoring`. Rebasing the other branch was turning into a mess.

Ensure all models in audit have Python 3-compatible `str()` methods using Django's `@python_2_unicode_compatible` decorator. Fixed a bug in `AuditorStaffMember`.

https://docs.djangoproject.com/en/1.11/ref/utils/#django.utils.encoding.python_2_unicode_compatible

Add non-ASCII tests for `str()` and `unicode()` methods for most audit models. 

In `EquiTrack/tests/test_all_models.py`, remove `audit` from the list of modules not tested for Python 3-compatibility since it's now compliant.

### CODE REVIEW

File: EquiTrack/audit/models.py [hunk 1]
```diff
@@ -45,10 +45,8 @@ class AuditorStaffMember(BaseStaffMember):
     auditor_firm = models.ForeignKey(AuditorFirm, verbose_name=_('firm'), related_name='staff_members')
 
     def __str__(self):
-        return '{} ({})'.format(
-            self.get_full_name(),
-            self.auditor_firm.name
-        )
+        auditor_firm_name = ' ({})'.format(self.auditor_firm.name) if hasattr(self, 'auditor_firm') else ''
```
[the remark this answers is not in the document]
Comment (caktus-philip) [reply EquiTrack/audit/models.py]: Prior to saving, `auditor_firm` can be None/NULL in which case any attempt to access it raises `RelatedObjectDoesNotExist`. My tests encounter that case because they use FactoryBoy's `build` strategy which doesn't save objects to the database. (That makes the tests run faster.) 

If you want to see the `RelatedObjectDoesNotExist` exception first hand, add a reference to `self.auditor_firm.name` to the top of `AuditorStaffMember.__str__()` and then run `audit.tests`.

**Comment by caktus-philip** [comment]
@domdinicola question answered; please re-review when you get a chance.

File: EquiTrack/audit/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (domdinicola) [reply EquiTrack/audit/models.py]: Ok clear! Thanks for explaining
